### DISCUSSION TITLE
How to pass <Chip> component onClick value's to my form data?

### DISCUSSION THREAD

**Discussion by AndresJaramillo95** [react-hook-form/react-hook-form#4307] [Q&A] [answered]
Greetings! 

Been struggling to try and solve this one,

I have a `<form>` that is wrapped around a Controller using a render. The render is a list of `<Chips>`'s. I want the list of chips to act as a field, each user toggled chip would act as the `data` inputted to the form, based on the label value of a chip. The expected data can be either one string or an array of strings based on how many `<Chips>` are toggled. 

I've tried to replicate examples that use `<Checkbox>`, but I wasn't able to re-create the component. 

I have a state of total chips toggled, which re-produces the data I would like to submit via my button. Is there a way to pass this state to my form data?

I'm currently only able to submit empty data through my form because my chip `values` and form `data` are not linked up correctly.

**Form**
```js
        <form noValidate  onSubmit = { handleSubmit(onSubmit) }>
            <Controller
                    render={({ onChange ,...props }) => (
                        <Paper component="ul" className={classes.root}>
                            {stocklist.map((value, index) =>
                            {
                            return (
                            <li key={index}>
                            <Chip
                            name="stock_list"
                            variant="outlined"
                            label={value}
                            key={index}
                            color={checked.includes(value) ? 'secondary' : 'primary'}
                            onClick={handleToggle(value)}
                            className={classes.chip}
                            ref={register}
                            />
                            </li> 
                            );
                            })}
                        </Paper>
                    )}
                    name="stock_list"
                    control={control}
                    defaultValue={[]}
                    onChange={([, data]) => data}
                    />
                {checked && checked.length > 0 &&
                    <Fab
                        color="primary"
                        aria-label="delete"
                        type="submit"
                    >
                        <DeleteIcon />
                    </Fab>
                }
        </form>
```

Here is how I toggle the chips, and create a state `checked` which holds the values for every toggled chip
```JS
   const { register, control, handleSubmit } = useForm();
   const [checked, setChecked] = React.useState([]);

    const handleToggle = (value) => () => {
        const currentIndex = checked.indexOf(value);
        const newChecked = [...checked];
     
        if (currentIndex === -1) {
            newChecked.push(value);
        } else {
            newChecked.splice(currentIndex, 1);
        }
     
        setChecked(newChecked);
        };

```

Here is my **onSubmit** function
```js
    const onSubmit = (data, e) =>
      {  
        console.log(data);
    
        axiosInstance
          .patch('url', {
            stock_list: data.stock_list,
          })
          .then((res) =>
          {
            console.log(res);
            console.log(res.data);
          });
    };

```
**Visual Aids**
Below is an example, as you see the values KHC and COST are toggled by the user
![image](https://user-images.githubusercontent.com/51006543/109397447-361e9e00-78fc-11eb-97ed-233f7324d055.png)

My console then logs the state for all toggled chips, which presents the ideal data I want to be submitting:
![image](https://user-images.githubusercontent.com/51006543/109397462-459de700-78fc-11eb-9b42-cd68764fc8c3.png)

Thank you to any hackers who can help me out on this one 👍

**Comment by bluebill1049** [comment]
Do you u have a simple codesandbox?

**Reply by AndresJaramillo95** [reply]
Hey @bluebill1049 , [Check it out here](https://codesandbox.io/s/cocky-roentgen-j0pcg). Please toggle one of the chips, and then press the button that appears. If you then check the console, you will notice an empty form array being submitted.

**Comment by bluebill1049** [comment, marked as answer]
<img width="512" alt="Screen Shot 2021-02-28 at 10 31 12 am" src="https://user-images.githubusercontent.com/10513364/109403243-15b01b80-79b0-11eb-9244-232cdbdea574.png">

Controller's `onChange` is never used.

**Reply by bluebill1049** [reply]
That's what send data to hook form.

**Reply by AndresJaramillo95** [reply]
Oh okay makes sense, here is how I fixed it.  :)

```js
    const handleToggle = (value) => {// <== I changed this part
        const currentIndex = checked.indexOf(value);
        const newChecked = [...checked];

        if (currentIndex === -1) {
          newChecked.push(value);
        } else {
          newChecked.splice(currentIndex, 1);
        }
        setChecked(newChecked);
        return newChecked;// <== I changed this part
      };
        console.log(checked);
```


```js
onClick={(e) => {
    onChange(handleToggle(value))
  }}
```

Thanks again for your timely responses!

**Reply by bluebill1049** [reply]
no worries! glad it's working out for you.
